src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -389,36 +389,42 @@ internal PSKeyword GetLogKeywords()
             if (!File.Exists(fileName)) { return defaultValue; }
 
             // Open file for reading, but allow multiple readers
-            fileLock.EnterReadLock();
-            try
+            if (fileLock.TryEnterReadLock(millisecondsTimeout: 10))
```

Comment:
>  With this context, is the TryEnterReadLock necessary?

We have race conditions in xUnit tests. It is related threads. And I am trying to address only the thread race conditions.

Reply:
You saw `EnterReadLock()` got blocked for a long time in xUnit test run? Do you still have the link to that CI run?
That's possible only if there are a lot write operations and each write operation takes a relatively long time. I doubt this would be possible in real world scenarios.

Reply:
There is some issues #8715 #8784 #8838 where we tracking xUnit test failures and there is links to CIs.

> That's possible only if there are a lot write operations and each write operation takes a relatively long time. I doubt this would be possible in real world scenarios.

If somebody open the config file with read lock PowerShell doesn't start.
Try:
```powershell
$path = "powershell.config.json "
$mode = "Open"
$access = "Read"
$share = "None"

$file = [System.IO.File]::Open($path, $mode, $access, $share)
```
then try to start new PowerShell Core. It is DoS attack.

Reply:
I looked at the issues you mentioned, and all are related to intra-process conflicts, not race conditions within a process.
For the DoS attack example, it's still an intra-process scenario, where `fileLock` doesn't play a role.

We discussed about this DoS attack in https://github.com/PowerShell/PowerShell/pull/8249#discussion_r236435798. Returning default value when the config file is not accessible will cause security settings to not be enforced, which would be another security hole. @TravisEz13 and I chatted about this offline. We have some thoughts but immature overall. Need to think about this problem more.

Reply:
Sure, let's focus on this PR specifically.

> My idea for the PR is to decrease file lock times internally in the process - it will reduce probability of intra-process locks and waitings too.

IMHO, I don't think `filelock` affects the race condition among processes, and making it timeout in 10 ms doesn't make sense given `waitfile` can block for 400 ms within the file lock.
I think for this PR, maybe the only change that should be done is to use `WaitForFile` in `UpdateValueInFile`.

Reply:
I replace fileLock.EnterReadLock() because it can fall in infinite wait. I do not know whether this can happen. Since testing race conditions is too complicated it will be more safe. I'd don't want to change current logic of the code until we push follow PR.

Reply:
> I replace fileLock.EnterReadLock() because it can fall in infinite wait

I would argue that the possibility is extremely slim unless we have data to prove otherwise.

Reply:
I rely on our experience. 
Reverted.